### PR TITLE
Added tilted_loss function used in quantile regression

### PR DESCRIPTION
**Pull request by sachinruk**
As name suggests, added the tilted loss function so that Keras may now do quantile regression. This means being able to calculate uncertainty estimates (or rather percentiles of observations to be exact). See my notebook and especially the last diagram to illustrate quantile regression: https://github.com/sachinruk/KerasQuantileModel/blob/master/Keras%20Quantile%20Model.ipynb

### CODE REVIEW

**Comment by fchollet** [comment]
This seems quite niche (not included in sklearn, for instance), so we won't include it in core Keras. You can try to submit this PR to keras-contrib instead.
